src/routes/docs/products/ai/integrations/togetherai/+page.markdoc

```diff
@@ -0,0 +1,326 @@
+  return res.json({ ok: false, error: 'Invalid field `type`' }, 400);
+}
+```
+
+In this example, you will use Mistral's Mixtral 8x7B for text generation and StabilityAI's Stable Diffusion XL for image generation. You can find more models on the [Together AI's docs](https://docs.together.ai/docs/inference-models).
+
+Next, add some per-model configurations:
+
+```js
+let request = {
+  model: models[req.body.type],
+};
+
+switch (req.body.type) {
+  case 'text':
+    request = {
+      ...request,
+      messages: [
+        {
+          role: "system",
+          content: "You are a helpful assistant"
+        },
+        {
+          role: "user",
+          content: req.body.prompt
+        }
+      ],
+      max_tokens: 512,
+      repetition_penalty: 1,
+    }
+  break;
+  case 'image':
+    request = {
+      ...request,
+      prompt: req.body.prompt,
+      width: 512,
+      height: 512,
+      steps: 20,
+      results: 1,
+      negative_prompt: "deformed, noisy, blurry, distorted",
+    }
+  break;
+};
+```
+
+This allows you to configure each of the models you use individually. Feel free to play with this configuration to get the best results for your use case.
+
+Finally, with the request built, you can call the Together AI API and generate a prediction:
+
+```js
+let response;
+let url = 'https://api.together.xyz/v1/completions';
+
+if (req.body.type === 'text') {
+  url = 'https://api.together.xyz/v1/chat/completions'
+};
+
+try {
+  response = await fetch(URL, {
+    headers: {
+      "content-type": "application/json",
+      "Authorization": `Bearer ${process.env.TOGETHER_API_KEY}`
+    },
+    method: 'POST',
+    body: JSON.stringify(request)
+  })
+} catch (err) {
+  error(err);
+  return res.json({ ok: false, error: 'Failed to run model' }, 500);
+}
+
+let resJson = await response.json();
+
+// Upload image to Appwrite Storage and return URL
+if (req.body.type === 'image') {
+  const endpoint = process.env.APPWRITE_ENDPOINT || 'https://cloud.appwrite.io/v1'
+
+  const client = new Client()
+    .setEndpoint(endpoint)
+    .setKey(process.env.APPWRITE_API_KEY)
+    .setProject(process.env.APPWRITE_FUNCTION_PROJECT_ID)
+
+  const storage = new Storage(client);
+
+  let data = Buffer.from(resJson.choices[0].image_base64, 'base64');
+
+  let file = await storage.createFile(process.env.APPWRITE_BUCKET_ID, ID.unique(), InputFile.fromBuffer(data, "image.png"));
+
+  return res.json({
+    ok: true,
+    type: req.body.type,
+    response: `${endpoint}/storage/buckets/${process.env.APPWRITE_BUCKET_ID}/files/${file["$id"]}/view?project=${process.env.APPWRITE_FUNCTION_PROJECT_ID}`
+  })
+}
+
+return res.json({ ok: true, type: req.body.type, response: resJson.choices[0].message.content}, 200);
+```
+
+This code sends the prompt to the Together AI API and returns the response to the user.
+The function also uploads any images generated to a bucket and returns the URL to the user.
```

Comment:
There is no mention of the variable `APPWRITE_BUCKET_ID` in the pre requisites section & step 5 of Section #1